src/sage/rings/finite_rings/integer_mod_ring.py

```diff
@@ -1937,10 +1965,35 @@ def _roots_univariate_polynomial(self, f, ring=None, multiplicities=True, algori
             # Roots of non-zero polynomial over finite fields by factorization
             return f.change_ring(f.base_ring().field()).roots(multiplicities=multiplicities)
 
-        # Zero polynomial is a base case
+        # Constant and linear polynomials are base cases
         if deg < 0:
             # All residue classes are roots of the zero polynomial
             return [*map(self, range(self.cardinality()))]
+        elif deg == 0:
+            return []
+        elif deg == 1:
+            # assume form a*x + b
+            b, a = f.list()
+            if a.is_unit():
+                return [-b * (~a)]
+            else:
+                al, bl = a.lift(), b.lift()
+
+                N = self.order()
+                g = N.gcd(al)
+
+                if bl % g != 0:
+                    return [] # No solution
+                else:
+                    # whole eqn divided by g
+                    _R = Zmod(N/g)
```

Comment:
You might want to use N // g by the way, else I think this will become a rational? (Of course it tries to coerce back to integer but slow)

Reply:
there's also `divide_knowing_divisible_by`.